src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -143,16 +151,18 @@ public ParentFieldMapper build(BuilderContext context) {
 
     private final String type;
     private final BytesRef typeAsBytes;
+    private final boolean docValuesJoin;
 
-    protected ParentFieldMapper(String name, String indexName, String type, @Nullable Settings fieldDataSettings, Settings indexSettings) {
+    protected ParentFieldMapper(String name, String indexName, String type, @Nullable Settings fieldDataSettings, Settings indexSettings, boolean docValuesJoin) {
```

Comment:
Instead of this being a boolean, can the mapper use the index settings as others do to decide its behavior (essentially move the check deciding on doc values based on version into the ctor here)?